src/shared/use-app-engine-middleware.js

```diff
@@ -19,11 +21,13 @@ export async function useAppEngineMiddleware<TReq: $Request, TRes: $Response>(
         express<TReq, TRes>()
             // Add the request logging middleware.
             .use(await makeRequestMiddleware(mode, logger))
-            // Add requestID middleware
+            // Add requestID middleware.
             .use(makeAppEngineRequestIDMiddleware(logger))
             // Add the app.
             .use(app)
             // Add the error logging middleware.
             .use(makeErrorMiddleware(logger))
+            // Add memory monitoring.
+            .use(asyncHandler(makeMemoryMonitoringMiddleware(logger)))
```

Comment:
Need to modify this bit to just not add it if the env vars aren't there at setup time. I'll make that change.